### ISSUE TITLE
false positive ts2310: Type '' recursively references itself as a base type. (regression 4.2.4 -> 4.3.2)

### ISSUE THREAD

**Issue by mrgleba** [microsoft/TypeScript#44359]
# Bug Report

### 🔎 Search Terms

ts2310 

### 🕗 Version & Regression Information

- This changed between versions 4.2.4 and 4.3.2

### ⏯ Playground Link

[Playground link](https://www.typescriptlang.org/play?ts=4.3.2#code/FAQwRgzgLgTiDGUAE8A2IISQIQwUyQG9glSlxo5EkYB7WqACgEokAuJAETxgEsA3PABNgAX2DA0GLNz6ChSPAA8oeAHZCsUzIpXrNOfERJk6DFsbJWaeKAFcYapHY14AZrzXDyWEGoCeJqTi4sRWYkA)

### 💻 Code

```ts

abstract class Base {
    abstract root() : Derived
}

class Derived extends class extends Base {
    root() {
        return undefined as any
    }
}
{
    
}
```

### 🙁 Actual behavior

Compiler shows ts2310 error

### 🙂 Expected behavior

Clean compile

**Comment by canonic-epicure** [comment]
Perhaps same as https://github.com/microsoft/TypeScript/issues/44281 ?

**Comment by mrgleba** [comment]
Thanks for pointing that out.
I'm closing the issue as a duplicate.
